boakfriends/hammeroids
# Remove player from lobby and broadcast on disconnect.

#### What's this PR do?
Refactors some the connection events out of the Subscription class and
back up in to the main Socket block, this makes it a bit clearer to see
the lifecycle of the connection.

When a connection disconnects the player is removed from the lobby and
updated lobby is broadcast to all connections in the channel.

##### Background context

Players were hanging around in the lobby after disconnect.

#### How should this be manually tested?

Join the game in two windows, see both players listed in lobby. Close one window and the lobby will update to just show the first player.

#### Screenshots
![lobby-leave](https://user-images.githubusercontent.com/847787/53897696-95f44e00-402e-11e9-8976-57af35e766f3.gif)
